graphql-server/pkg/dataloaders/proposal.go

```diff
@@ -22,9 +24,36 @@ type ProposalTallyResultDataloader interface {
 	LoadAll(ids []int) ([]*models.ProposalTallyResult, []error)
 }
 
+type ProposalVoteKey struct {
+	ProposalID int
+	Address    string
+}
+type ProposalVoteDataloader interface {
+	Load(key ProposalVoteKey) (*models.ProposalVote, error)
+	LoadAll(keys []ProposalVoteKey) ([]*models.ProposalVote, []error)
+}
+
+type ProposalDepositKey = ProposalVoteKey
+type ProposalDepositDataloader interface {
+	Load(key ProposalDepositKey) (*models.ProposalDeposit, error)
+	LoadAll(keys []ProposalDepositKey) ([]*models.ProposalDeposit, []error)
+}
+
 type IProposalDataloader struct {
 	proposalLoader            ProposalLoader
 	proposalTallyResultLoader ProposalTallyResultDataloader
+	proposalVoteLoader        ProposalVoteDataloader
+	proposalDepositLoader     ProposalDepositDataloader
+}
+
+// check if element is contained in array
+func contains[T comparable](arr []T, element T) bool {
+	for _, s := range arr {
+		if s == element {
+			return true
+		}
+	}
+	return false
 }
```

Comment:
should be in utils

Reply:
as discussed offline, will use https://pkg.go.dev/golang.org/x/exp@v0.0.0-20220613132600-b0d781184e0d/slices#Contains instead